### ISSUE TITLE
Reproduce the problem: Frontier emptied. Returning best known circuit with 0 layers and cost: 2.500003e-01.

### ISSUE THREAD

**Issue by jinleic** [BQSKit/bqskit#204]
This file can reproduce the error.

[knn_n41_compiled.txt](https://github.com/BQSKit/bqskit/files/13323978/knn_n41_compiled.txt)

It is a sort circuit with many ccx gate, which might makes it hard to partition.

**Comment by edyounis** [comment]
Can you check if the 1.1.1 branch fixes this?

**Comment by jinleic** [comment]
Yes, it fixes the problem with compiled knn_n41_compiled.txt where the gate set is {cx,sx,x,rz}

However, if a raw knn_n41.qasm is used as input, it still pops the frontier bug.
[knn_n41.txt](https://github.com/BQSKit/bqskit/files/13578101/knn_n41.txt)

This raw knn_n41.qasm contains CSWAP gates.
I noticed that the line to fix the problem is 

                if circuit.num_qudits != 2:
                    # Guard on >2 qubit to prevent high-error glitches
                    continue

Maybe a three-qubit gate is handled by this line.

**Comment by edyounis** [comment]
As discussed, this has been fixed with the recent 1.1.1 branch. Thanks for the bug report
